activerecord/test/cases/serialization_test.rb

```diff
@@ -103,4 +103,8 @@ def test_find_records_by_serialized_attributes_through_join
 
     assert_equal 1, Author.joins(:serialized_posts).where(name: "David", serialized_posts: { title: "Hello" }).length
   end
+
+  def test_serializes_active_record_constant
+    assert_equal Author.name, Author.as_json
```

Comment:
So, if your changes are to be correct, wouldn't you need to load the `Author.arel_table` first to ensure correctness?

Also, if I run `.arel_table` on a class before loading the arel table, I get something that looks like this:

```
2.4.0 :001 > Category.as_json
 => {"generated_attribute_methods"=>{"_mutex"=>{}}, "attribute_methods_generated"=>false, "generated_association_methods"=>{}, "arel_engine"=>nil, "arel_table"=>nil, "column_names"=>nil, "attribute_types"=>nil, "content_columns"=>nil, "default_attributes"=>nil, "inheritance_column"=>nil, "attributes_builder"=>nil, "columns"=>nil, "columns_hash"=>nil, "attribute_names"=>nil, "yaml_encoder"=>nil, "_type_candidates_cache"=>{"backend"=>{}, "write_lock"=>{}, "default_proc"=>nil}, "find_by_statement_cache"=>{"true"=>{}, "false"=>{}}, "relation_delegate_cache"=>{"ActiveRecord::Relation"=>{"delegation_mutex"=>{}}, "ActiveRecord::Associations::CollectionProxy"=>{"delegation_mutex"=>{}}, "ActiveRecord::AssociationRelation"=>{"delegation_mutex"=>{}}}}
```